packages/walletconnect/src/provider.ts

```diff
@@ -369,7 +370,11 @@ export function parseChain(chainString: string): ChainInfo {
   if (chainGroupDecoded < -1) {
     throw Error('Chain group in protocol needs to be either -1 or non-negative')
   }
-  return { networkId: networkId, chainGroup: chainGroupDecoded === -1 ? undefined : chainGroupDecoded }
+  const networkIdList = networkIds as ReadonlyArray<string>
+  if (!networkIdList.includes(networkId)) {
```

Comment:
nit: I think it's cleaner if you do it like this:
```ts
!networkIds.some((id) => id === networkId)
```

no need to do type casting then!